core/src/subgraph/registrar.rs

```diff
@@ -668,10 +679,79 @@ fn create_subgraph_version(
         },
     });
 
+    let start_blocks: Option<Vec<EthereumBlockPointer>> = manifest
+        .data_sources
+        .clone()
+        .into_iter()
+        .map(|data_source| {
+            let source_address = data_source.source.address;
+            let block_number = match data_source
+                .source
+                .start_block
+                .and_then(|block_str| block_str.parse::<u64>().ok())
+            {
+                Some(number) => number,
+                None => {
+                    warn!(
+                        logger,
+                        "Start block invalid or not specified for the '{}' data source",
+                        &data_source.name
+                    );
+                    return None
+                },
+            };
+
+            // Validate that the start block provided for the Data Source is indeed the block that
+            // contains the contract creation transaction. Allow the block to be used as the
+            // subgraph level start block regardless, but emit a warning.
+            match ethereum_adapter
+                .validate_start_block(logger, block_number, source_address)
+                .wait()
```

Comment:
Don't use `.wait()`, it tends to hang. Turning the whole function into a future might is annoying but we'll have to do it if there's no other way of avoiding `.wait()`.